y-board-v3/hardware_test/src/main.cpp

```diff
@@ -1,79 +1,251 @@
+#define REFRESH_RATE 50       // Measured in ms
+#define SCREEN_WIDTH 128
+#define ZERO_DEG 0
+#define PADDING 2 
+#define SWITCH_X 0+PADDING*2
+#define SWITCH_Y 8
+#define SWITCH_WIDTH 12
+#define SWITCH_HEIGHT 24
+#define BUTTON_Y SWITCH_Y+5
+#define BUTTON_WIDTH 17
+#define BUTTON_HEIGHT 17
+#define BUTTON_X SCREEN_WIDTH/2-BUTTON_WIDTH
+#define KNOB_RADIUS 8
+#define KNOB_X SCREEN_WIDTH-KNOB_RADIUS-PADDING*2
+#define KNOB_Y 20
+
+#define ON 1
+#define OFF 0
+
+Adafruit_SSD1306 display(128, 32); // Create display
+int x,y,z,but1,but2,sw1,sw2,knob,temp,knob_x,knob_y;
+
 void setup() {
-    Serial.begin(9600);
-    Yboard.setup();
+  Serial.begin(9600);
+  Yboard.setup();
+  x = 0;
+  y = 0;
+  z = 0;
+  but1 = Yboard.get_button(1);
+  but2 = Yboard.get_button(2);
+  sw1 = Yboard.get_switch(1);
+  sw2 = Yboard.get_switch(2);
+  knob = map(Yboard.get_knob(), 0, 100, 0, 255);
+  knob_x = cos(knob * PI / 180 + PI / 4) * KNOB_RADIUS + KNOB_X;
+  knob_y = sin(knob * PI / 180 + PI / 4) * KNOB_RADIUS + KNOB_Y;
+  temp = 0;
+
+  // Display setup
+  delay(1000); // Display needs time to initialize
+  display.begin(SSD1306_SWITCHCAPVCC,
+                0x3c); // Initialize display with I2C address: 0x3C
+  display.clearDisplay();
+  display.setTextColor(ON,OFF);
+  display.setRotation(ZERO_DEG); // Can be 0, 90, 180, or 270
+  display.setTextWrap(false);
+  uint8_t text_size = 1;
+  display.setTextSize(text_size);
+  display.setCursor(0, 0);
+  display.printf("x:%i",x);
+  display.setCursor(42, 0);
+  display.printf("y:%i", y);
+  display.setCursor(85, 0);
+  display.printf("z:%i", z);
+  display.drawRoundRect(SWITCH_X, SWITCH_Y, SWITCH_WIDTH, SWITCH_HEIGHT, 3, ON);
+  if (sw1) {
+    display.fillRoundRect(SWITCH_X+PADDING, SWITCH_Y+PADDING, 8, 8, 3, ON);
+  } else {
+    display.fillRoundRect(SWITCH_X+PADDING, SWITCH_Y+PADDING+SWITCH_HEIGHT/2, 8, 8, 3, ON);
+  }
+  display.drawRoundRect(SWITCH_X+SWITCH_WIDTH+PADDING, SWITCH_Y,SWITCH_WIDTH, SWITCH_HEIGHT, 3, ON);
+  if (sw2) {
+    display.fillRoundRect(SWITCH_X+SWITCH_WIDTH+PADDING*2, SWITCH_Y+PADDING, 8, 8, 3, ON);
+  } else {
+    display.fillRoundRect(SWITCH_X+SWITCH_WIDTH+PADDING*2, SWITCH_Y+PADDING+SWITCH_HEIGHT/2, 8, 8, 3, ON);
+  }
+  display.drawRect(BUTTON_X, BUTTON_Y, BUTTON_WIDTH, BUTTON_HEIGHT, ON);
+  if (but1) {
+    display.fillCircle(BUTTON_X+BUTTON_WIDTH/2, BUTTON_Y+BUTTON_HEIGHT/2-1, 6, ON);
+  }else {
+    display.drawCircle(BUTTON_X+BUTTON_WIDTH/2, BUTTON_Y+BUTTON_HEIGHT/2, 6, ON);
+  }
+  display.drawRect(BUTTON_X+BUTTON_WIDTH+PADDING, BUTTON_Y, BUTTON_WIDTH, BUTTON_HEIGHT, ON);
+  if (but2) {
+    display.fillCircle(BUTTON_X+BUTTON_WIDTH+PADDING+BUTTON_WIDTH/2, BUTTON_Y+BUTTON_HEIGHT/2, 6, ON);
+  } else {
+    display.drawCircle(BUTTON_X+BUTTON_WIDTH+PADDING+BUTTON_WIDTH/2, BUTTON_Y+BUTTON_HEIGHT/2, 6, ON);
+  }
+  display.drawCircle(KNOB_X, KNOB_Y, KNOB_RADIUS, ON);
+  display.display();
+  Yboard.play_notes("O5 F D. C#8 D. C#8 D8 B8- O4 G8 F"); // Play fight song

```

Comment:
Instead of playing the song in the setup, can you play it when button one is pressed? Currently, it plays some other notes.